_ci/verify-release-assets.sh

```diff
@@ -0,0 +1,74 @@
+#!/bin/bash
+# Script to verify the release assets by downloading them from the GitHub release
+# and checking if they are accessible. If the asset is not accessible, the script
+# will try to delete asset and re-upload it to the release.
+# https://github.com/gruntwork-io/terragrunt/issues/3220
+
+# Ensure CIRCLE_TAG and GITHUB_OAUTH_TOKEN are set
+if [ -z "$CIRCLE_TAG" ]; then
+  echo "CIRCLE_TAG environment variable is not set. Exiting..."
+  exit 1
+fi
+
+if [ -z "$GITHUB_OAUTH_TOKEN" ]; then
+  echo "GITHUB_OAUTH_TOKEN environment variable is not set. Exiting..."
+  exit 1
+fi
+
+RELEASE_TAG=$CIRCLE_TAG
+REPO_OWNER="gruntwork-io"
+REPO_NAME="terragrunt"
+MAX_RETRIES=10
+
+RELEASE_RESPONSE=$(curl -s \
+  -H "Accept: application/vnd.github.v3+json" \
+  -H "Authorization: token $GITHUB_OAUTH_TOKEN" \
+  "https://api.github.com/repos/$REPO_OWNER/$REPO_NAME/releases/tags/$RELEASE_TAG")
+
+# Check if the release exists
+if echo "$RELEASE_RESPONSE" | jq -e '.message == "Not Found"' > /dev/null; then
+  echo "Release $RELEASE_TAG not found. Exiting..."
+  exit 1
+fi
+
+# Get the release id
+RELEASE_ID=$(echo "$RELEASE_RESPONSE" | jq -r '.id')
+ASSET_URLS=$(echo "$RELEASE_RESPONSE" | jq -r '.assets[].browser_download_url')
+
+# Loop through each asset URL and attempt to download
+for ASSET_URL in $ASSET_URLS; do
+  ASSET_NAME=$(basename "$ASSET_URL")
+
+  for ((i=0; i<MAX_RETRIES; i++)); do
+    if ! curl -sILf "$ASSET_URL" > /dev/null; then
+      echo "Failed to download the asset $ASSET_NAME. Retrying..."
+
+      # Delete the asset
+      ASSET_ID=$(echo "$RELEASE_RESPONSE" | jq -r ".assets[] | select(.name==\"$ASSET_NAME\") | .id")
```

Comment:
NIT: It seems like it wasn't an issue here, but it's generally better to use the `--arg` flag of JQ for parameters.